cdk-testsuite/tests/openshiftSanity.py

```diff
@@ -35,39 +37,120 @@ def setUp(self):
                               self.params.get('openshift_PASSWORD'), self.params.get('openshift_ruby_PROJECT'), 
                               self.params.get('openshift_ruby_REGISTRY'), self.params.get('service_ruby_NAME'))
     
     def test_perl_project(self):
         '''
-        TBD
+        Runs sanity on openshift s2i perl source
+            1. login
+            2. new project create
+            3. App deploy using the registry from config file
+            4. Validation for successful deployment
+            5. Routing
+            6. Pod status
+            7. project delete
+            8. logout
+        Args:
+            self (object): Object of the current method
+            openshift_URL (string) : ip and port of openshift server to be used
+            openshift_USERNAME (string): username of openshift web console to be used
+            openshift_PASSWORD (string): password of openshift web console to be used
+            openshift_perl_PROJECT (string): name of the perl project to be added to the openshift server
+            openshift_perl_REGISTRY (string): perl registry path/location
+            service_perl_NAME (string): name of the perl service to be exposed outside
         '''
         openshift.new_project(self, self.params.get('openshift_URL'), self.params.get('openshift_USERNAME'), 
                               self.params.get('openshift_PASSWORD'), self.params.get('openshift_perl_PROJECT'), 
                               self.params.get('openshift_perl_REGISTRY'), self.params.get('service_perl_NAME'))
     
     def test_nodejs_project(self):
         '''
-        TBD
+        Runs sanity on openshift s2i nodejs source
+            1. login
+            2. new project create
+            3. App deploy using the registry from config file
+            4. Validation for successful deployment
+            5. Routing
+            6. Pod status
+            7. project delete
+            8. logout
+        Args:
+            self (object): Object of the current method
+            openshift_URL (string) : ip and port of openshift server to be used
+            openshift_USERNAME (string): username of openshift web console to be used
+            openshift_PASSWORD (string): password of openshift web console to be used
+            openshift_nodejs_PROJECT (string): name of the nodejs project to be added to the openshift server
+            openshift_nodejs_REGISTRY (string): nodejs registry path/location
+            service_nodejs_NAME (string): name of the nodejs service to be exposed outside
         '''
         openshift.new_project(self, self.params.get('openshift_URL'), self.params.get('openshift_USERNAME'), 
                               self.params.get('openshift_PASSWORD'), self.params.get('openshift_nodejs_PROJECT'), 
                               self.params.get('openshift_nodejs_REGISTRY'), self.params.get('service_nodejs_NAME'))
     
     def test_php_project(self):
         '''
```

Comment:
All these functions look the same. Why not have a generic function by passing application details as an argument. The parameters for the application could be passed in a dict too. This will help us run any application.